### PR TITLE
warn if self-closing tag is used

### PR DESCRIPTION
**Pull request by tranvansang**
The edit is not well tested. I edited it directly in github

### CODE REVIEW

File: packages/intl-messageformat/src/formatters.ts [hunk 1]
```diff
@@ -361,6 +361,11 @@ export function formatHTMLMessage(
 
       tagName = tagName.toLowerCase();
 
+      if (process.env.NODE_ENV !== 'production' && [
+        'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'link', 'meta', 'param', 'source', 'track', 'wbr'
+      ].includes(tagName))
+        console.warn(`${tagName} is a self-closing tag and can not be used, please use another tag name.`)
```
Comment (longlho) [inline packages/intl-messageformat/src/formatters.ts]: u should just `throw` a `FormatError` cause it'll produce invalid output (swallow children) which is even harder to debug

**Review by longlho — CHANGES_REQUESTED** [review]

File: packages/intl-messageformat/src/formatters.ts [hunk 2]
```diff
@@ -361,6 +361,11 @@ export function formatHTMLMessage(
 
       tagName = tagName.toLowerCase();
 
+      if (process.env.NODE_ENV !== 'production' && [
```
Comment (longlho) [inline packages/intl-messageformat/src/formatters.ts]: I think u can drop `process.env.NODE_ENV` check as well

File: packages/intl-messageformat/src/formatters.ts [hunk 3]
```diff
@@ -361,6 +361,11 @@ export function formatHTMLMessage(
...
+        'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'link', 'meta', 'param', 'source', 'track', 'wbr'
+      ].includes(tagName))
```
Comment (longlho) [inline packages/intl-messageformat/src/formatters.ts]: `~[].indexOf()` bc IE11 (& we don't package w/ any polyfill due to weight)

File: packages/intl-messageformat/src/formatters.ts [hunk 4]
```diff
@@ -361,6 +361,11 @@ export function formatHTMLMessage(
...
+        throw new Error(`${tagName} is a self-closing tag and can not be used, please use another tag name.`)
```
Comment (longlho) [inline packages/intl-messageformat/src/formatters.ts]: `throw new FormatError`

**Review by longlho — CHANGES_REQUESTED** [review]

**Review by longlho — APPROVED** [review]

**Comment by longlho** [comment]
Thanks a lot for your contributions!

**Comment by klimashkin** [comment]
Is there a `why` ticket for these changes?

**Comment by longlho** [comment]
context: 
https://github.com/formatjs/formatjs/issues/152
https://github.com/formatjs/formatjs/issues/148

This is to prevent footguns where you can specify `<br>blah</br>` and have `blah` be swallowed by `DOMParser`.

We've added caveats here: https://github.com/formatjs/formatjs/tree/master/packages/intl-messageformat#caveats

**Pull request merged**